lib/Parse/ParseDecl.cpp

```diff
@@ -1460,6 +1347,41 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
         .fixItReplace(Tok.getLoc(), "available");
   }
 
+
+  if (DK == DAK_Count && Tok.getText() == "warn_unused_result") {
```

Comment:
Keeping the attribute kind in the list causes it to have to show up in a lot of places in several files. It also still requires actually parsing it, validating the args, and keeping all of the associated information. I did that at first, and I found it to be much uglier. 

Also I believe this is the first case after `availability` where an attribute was totally removed instead of just renaming it or moving it to a new position so it seemed to belong here.